File: app/src/main/java/com/wr15/mytestproject/adapter/HomeMovieAdapter.java
package com.wr15.mytestproject.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.android.volley.RequestQueue;
import com.bumptech.glide.Glide;
import com.wr15.mytestproject.R;
import com.wr15.mytestproject.model.MFilm;
import com.wr15.mytestproject.server.ServerMovieHost;

import java.util.List;

public class HomeMovieAdapter extends RecyclerView.Adapter<HomeMovieAdapter.Holder> {

    private List<MFilm> mfilms;
    private ItemClickListener<MFilm> listenermfilm;

    private String TAG = HomeMovieAdapter.class.getSimpleName();

    private RequestQueue requestQueue;

    private Context mcontext;

    private double Rating;


    public HomeMovieAdapter(Context context, List<MFilm> Mfilms) {
        this.mfilms = Mfilms;
        this.mcontext = context;
    }


    public void setListener(ItemClickListener listenerMfilm) {
        this.listenermfilm = listenerMfilm;
    }

    @Override
    public Holder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new Holder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_home_movie, parent, false));
    }


    @Override
    public void onBindViewHolder(HomeMovieAdapter.Holder holder, final int position) {

        final MFilm data_film = mfilms.get(position);

        Glide.with(holder.foto.getContext())
                .load(ServerMovieHost.URLIMAGE + data_film.getPoster_path())
                .error(R.drawable.ic_baseline_movie_24)
                .placeholder(R.drawable.ic_baseline_movie_24)
                .into(holder.foto);

        holder.nama.setText(data_film.getTitle());

        Rating = data_film.getVote_average();

        float newValue = (float)Rating;
        holder.ratingBar.setNumStars(5);
        holder.ratingBar.setStepSize((float) 0.5);
        holder.ratingBar.setRating(newValue / 2);




        holder.item.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listenermfilm.onClicked(mfilms.get(position), position, v);



            }
        });


    }

    @Override
    public int getItemCount() {
        return mfilms.size();
    }

    class Holder extends RecyclerView.ViewHolder {

        private TextView nama;
        private ImageView foto;
        private LinearLayout item;
        private RatingBar ratingBar;

        public Holder(View itemView) {
            super(itemView);
            item = itemView.findViewById(R.id.linearFilm);
            nama = itemView.findViewById(R.id.txtNamaFilm);
            foto = (ImageView) itemView.findViewById(R.id.imgFilm);

            ratingBar = (RatingBar) itemView.findViewById(R.id.rtFilm);

        }

    }



}

File: app/src/main/java/com/wr15/mytestproject/adapter/HomeMovieAdapter.kt
package com.wr15.mytestproject.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.RatingBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.RequestQueue
import com.bumptech.glide.Glide
import com.wr15.mytestproject.R
import com.wr15.mytestproject.model.MFilm
import com.wr15.mytestproject.server.ServerMovieHost

class HomeMovieAdapter(private val mcontext: Context, private val mfilms: List<MFilm>) :
    RecyclerView.Adapter<HomeMovieAdapter.Holder>() {
    private var listenermfilm: ItemClickListener<MFilm>? = null
    private val TAG = HomeMovieAdapter::class.java.simpleName
    private val requestQueue: RequestQueue? = null
    private var Rating = 0.0
    fun setListener(listenerMfilm: ItemClickListener<*>?) {
        listenermfilm = listenerMfilm as ItemClickListener<MFilm>?
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        return Holder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_home_movie, parent, false)
        )
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val data_film = mfilms[position]
        Glide.with(holder.foto.context)
            .load(ServerMovieHost.URLIMAGE + data_film.poster_path)
            .error(R.drawable.ic_baseline_movie_24)
            .placeholder(R.drawable.ic_baseline_movie_24)
            .into(holder.foto)
        holder.nama.text = data_film.title
        Rating = data_film.vote_average
        val newValue = Rating.toFloat()
        holder.ratingBar.numStars = 5
        holder.ratingBar.stepSize = 0.5.toFloat()
        holder.ratingBar.rating = newValue / 2
        holder.item.setOnClickListener(View.OnClickListener { v ->
            listenermfilm!!.onClicked(
                mfilms[position], position, v
            )
        })
    }

    override fun getItemCount(): Int {
        return mfilms.size
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nama: TextView
        val foto: ImageView
        val item: LinearLayout
        val ratingBar: RatingBar

        init {
            item = itemView.findViewById(R.id.linearFilm)
            nama = itemView.findViewById(R.id.txtNamaFilm)
            foto = itemView.findViewById<View>(R.id.imgFilm) as ImageView
            ratingBar = itemView.findViewById<View>(R.id.rtFilm) as RatingBar
        }
    }
}